src/Utility/request-generator-replacers.js

```diff
@@ -42,7 +42,13 @@ const replacer_factory = that => ({
             id="new-request-button"
             onClick={() => {
                 if (!that.state.request.done) that.confirmNewRequest();
-                else that.newRequest();
+                else
+                    that.newRequest().then(() => {
+                        // We are in batch mode, move to the next company.
+                        if (that.state.batch && that.state.batch.length > 0) {
+                            that.setCompanyBySlug(that.state.batch.shift()).then(that.renderLetter);
+                        } else that.renderLetter();
+                    });
```

Comment:
This should be wrapped in curly braces. But in the interest of getting this important fix merged quickly, I'll let that slide…